include/ibamr/IBFEInstrumentPanel.h

```diff
@@ -0,0 +1,252 @@
+     * in the parent mesh.
+     * dimension 1 = number of meter meshes
+     * dimension 2 = number of mesh nodes
+     * dimension 3 = NDIM
+     */
+    std::vector<std::vector<std::vector<libMesh::dof_id_type> > > d_U_dof_idx;
+    std::vector<std::vector<std::vector<libMesh::dof_id_type> > > d_dX_dof_idx;
+
+    /*!
+     * \brief a vector containing the nodes of each meter mesh.
+     */
+    std::vector<std::vector<libMesh::Point> > d_nodes;
+
+    /*!
+     * \brief a vector storing the dof indices for each meter mesh.
+     */
+    std::vector<std::vector<libMesh::dof_id_type> > d_node_dof_IDs;
+
+    /*!
+     * \brief contains pointers to the equation systems for the meter mesh.
+     */
+    std::vector<libMesh::EquationSystems*> d_meter_systems;
+
+    /*!
+     * \brief vector of exodus io objects for data output.
+     */
+    std::vector<libMesh::ExodusII_IO*> d_exodus_io;
+
+    /*!
+     * \brief vector of meter mesh pointers.
+     */
+    std::vector<libMesh::SerialMesh*> d_meter_meshes;
+
+    /*!
+     * \brief names for each meter mesh.
+     */
+    std::vector<std::string> d_meter_mesh_names;
+
+    /*!
+     * \brief contains the nodeset IDs on the parent mesh, for the nodesets
+     * from which the meter meshes are built.
+     */
+    SAMRAI::tbox::Array<int> d_nodeset_IDs_for_meters;
+
+    /*!
+     * \brief things for data io.
+     */
+    int d_instrument_dump_interval;
+    std::vector<double> d_flow_values, d_mean_pressure_values;
+    std::string d_plot_directory_name;
+    std::ofstream d_mean_pressure_stream;
+    std::ofstream d_flux_stream;
+
+    /*!
+     * \brief for ordering objects in a multimap.
+     */
+    struct IndexFortranOrder : public std::binary_function<SAMRAI::hier::Index<NDIM>, SAMRAI::hier::Index<NDIM>, bool>
+    {
+        inline bool operator()(const SAMRAI::hier::Index<NDIM>& lhs, const SAMRAI::hier::Index<NDIM>& rhs) const
+        {
+            return (lhs(0) < rhs(0)
+#if (NDIM > 1)
+                    ||
+                    (lhs(0) == rhs(0) && lhs(1) < rhs(1))
+#if (NDIM > 2)
+                    ||
+                    (lhs(0) == rhs(0) && lhs(1) == rhs(1) && lhs(2) < rhs(2))
+#endif
+#endif
+                        );
+        }
+    };
+
+    /*!
+     * \brief struct for storing information about the quadrature points.
+     */
+    struct QuadPointStruct
+    {
+        int meter_num;               // meter ID
+        IBTK::Vector normal;         // normal vector at the quadrature point
+        IBTK::Vector qp_xyz_current; // current physical location of quadrature point
+        double JxW;                  // Jacobian multiplied by the quadrature weight
+    };
+
+    /*!
+     * \brief a multimap which associates SAMRAI indices with quadrature point structures.
+     */
+    typedef std::multimap<SAMRAI::hier::Index<NDIM>, QuadPointStruct, IndexFortranOrder> QuadPointMap;
+    std::vector<QuadPointMap> d_quad_point_map;
+};
+
+#endif /* IBFEINSTRUMENTPANEL_H */
```

Comment:
Sorry to be a broken record, but please see the end of IBInstrumentPanel.h for formatting...